assignments/array-methods.js

```diff
@@ -55,29 +55,73 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
-let fullName = [];
+let fullName = 
+    runners.forEach(function(firstlast) {
+return firstlast.first_name.last_name })
+
+
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
+ 
+ let allCaps = runners.map(function(runner) {
+    return runner.first_name.toUpperCase()
+})
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
-console.log(largeShirts);
+//let largeShirts = [];
+ let largeShirts = runners.filter(function(shirtsize) {
+    
+    return shirtsize.shirt_size === "L";
+    
+})
 
+console.log(largeShirts);
+//const reduceStatePopulations = data.reduce((total, state) => {
+ //   return total += state.population;
+ // }, 0);
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+//let ticketPriceTotal = [];
+let ticketPriceTotal = runners.reduce(function(total, Dons) {
+    return total += Dons.donation / 50;
+},0)
```

Comment:
Great idea to calculate to average here. You could make this even more general by using `runners.length` instead of hard coding `50`.